datalad/metadata/tests/test_aggregation.py

```diff
@@ -340,8 +340,8 @@ def test_partial_aggregation(path):
     # from-scratch aggregation kills datasets that where not listed
     ds.aggregate_metadata(path='sub1', incremental=False)
     res = ds.metadata(get_aggregates=True)
-    assert_result_count(res, 2)
-    assert_result_count(res, 0, path=sub2.path)
+    assert_result_count(res, 3)
+    assert_result_count(res, 1, path=sub2.path)
```

Comment:
This is a change in behavior. Not sure if good or bad. In general, I don't see any sense in throwing away metadata records because a dataset isn't present locally.